src/main/java/uk/ac/ebi/eva/commons/models/mongo/entity/subdocuments/VariantAnnotation.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.commons.models.mongo.entity.subdocuments;
+
+import org.springframework.data.mongodb.core.mapping.Field;
+import org.springframework.util.Assert;
+import uk.ac.ebi.eva.commons.models.mongo.entity.Annotation;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Lite version of genomic variant annotation generated using Ensembl VEP for indexing purposes.
+ */
+public class VariantAnnotation {
+
+    public static final String VEP_VERSION_FIELD = "vepv";
+
+    public static final String VEP_CACHE_VERSION_FIELD = "cachev";
+
+    public static final String SIFT_FIELD = "sift";
+
+    public static final String POLYPHEN_FIELD = "polyphen";
+
+    public static final String SO_ACCESSION_FIELD = "so";
+
+    public static final String XREFS_FIELD = "xrefs";
+
+    @Field(value = VEP_VERSION_FIELD)
+    private String vepVersion;
+
+    @Field(value = VEP_CACHE_VERSION_FIELD)
+    private String vepCacheVersion;
+
+    @Field(value = SIFT_FIELD)
+    private List<Double> sifts;
+
+    @Field(value = POLYPHEN_FIELD)
+    private List<Double> polyphens;
+
+    @Field(value = SO_ACCESSION_FIELD)
+    private Set<Integer> soAccessions = new HashSet<>();
+
+    @Field(value = XREFS_FIELD)
+    private Set<String> xrefIds = new HashSet<>();
+
+    VariantAnnotation() {
+        // Spring empty constructor
+    }
+
+    /**
+     * Variant annotation constructor. Requires non empty values, otherwise throws {@link IllegalArgumentException}
+     *
+     * @param vepVersion
+     * @param vepCacheVersion
+     * @throws IllegalArgumentException
```

Comment:
Information about parameters, return values and exceptions is expected in the corresponding tags. We should use those appropriately from now on and, in those cases where that rule is not followed, make changes.